### ISSUE TITLE
Frequent 500 Internal Server Error & significant drop in analytics.

### ISSUE THREAD

**Issue by mjdwebdesign** [rainlab/blog-plugin#300]
Hi,

I'm currently in an on-going battle with a frequent 500 internal server error.
I know that before implementing OctoberCMS we never had this issue as it was just a standard site, no fancy databases or scripts running.

Could you please let me know how this can be resolved. Also, whether there is anything I can provide that will aid in finding a solution to this problem.

--

As well that, we've encountered an extremely confusing change in our analytics and traffic flow ever since implementing OctoberCMS.

Here's our traffic overview across a 120 day period.

![image](https://cloud.githubusercontent.com/assets/23475543/24192373/9a02c20e-0ee6-11e7-8315-817cbd1900c9.png)

The area with practically 0 stats is when we actually changed over to the new website using OctoberCMS. That was simply because I forgot to implement the Google Analytics script onto the layouts of the site so, it didn't track anything. That's no problem.

Previously, before implementing OctoberCMS we peaked at 140 views in a single day.
Now, after implementing OctoberCMS we only seem to peak at 39.

Myself and the media coordinator I work with are both as confused as each other at this moment in time. We can't seem to pinpoint where or how this change has occurred.

The site is mobile friendly so, there are no issues regarding the site now being incompatible with certain devices. Although the new website involved a new design and maybe change of a few domain paths here and there, we can't figure out what is going on.

The initial reason for implementing OctoberCMS was to include a newsfeed which, is what we did.
So if anything, we've improved and added to the website so we were totally expecting an increase in traffic overflow.

Is there anything at all that I've missed out during the implementation of OctoberCMS or is there something we're missing in general that is causing this sudden decrease in traffic flow.

We're extremely confused as to the cause of the problem, how simply refreshing the design and implementing a newsfeed can stop as many as 100 views per day coming through.

We appreciate any potential solutions to these problems.
Thanks in advance,
Matt.

**Comment by mjdwebdesign** [comment]
Apologies.

I posted this issue in the blog plugin Github without even thinking about it.
I've now looking at creating a new issue over on the OctoberCMS Github page.

**Comment by LukeTowers** [comment]
Lots of factors can be influencing the change in viewers. I would take a look at those previous viewers, find out what they were actually doing on the site as far as what pages they were visiting. It's possible that a) most of them were bots anyways or b) not all of the content was transferred over smoothly so maybe now you have more 404 errors popping up or c) your site is loading slower now because you don't have adequate hosting in place or d) literally any reason whatsoever - a lot of factors affect site traffic.

I don't think either repository is the place to log this as an issue, issues are for reporting bugs and problems with the code, not for complaining that you've been noticing less visitors to your site.

**Comment by LukeTowers** [comment]
As far as the internal server errors go, enable debug mode and get the full error message, otherwise nobody's going to be able to help you.

**Comment by mjdwebdesign** [comment]
Hi Luke,

Thanks for your reply.

I understand what you're saying about this not being the correct place to ask about poor traffic. I have been meaning to create an issue regarding the 500 Internal Server errors we're getting anyway so, I thought best just to include it in with that to see if someone could shed light on our situation.

It's not bots. We're genuinely a fairly large organisation, working with the community side of a professional football club. The traffic is genuine.

I'm very much still in the learning process of using OctoberCMS and content management systems in general so I didn't even think to check the request logs, until I noticed there were 1,900+ of them.

I think I've now found the solution as to why our analytics are showing lower numbers. Before implementing OctoberCMS, the old website was simply standard HTML, no fancy scripts, server-side or client-side.

I believe a lot of visitors are still trying to access pages like so: ~/index.html
Of course, these pages now lie at: ~/index.php

I'm now in the process of slowly but surely sifting through these request logs to resolve them as best I can. At the moment, I've turned to redirecting visitors to the relevant, correct version of the page that would have existed previously.

The majority of these request logs consist of the exact same URLs however, visitors trying to access ~/index.html instead of ~/index.php.

Do you have any suggestions on how I could potentially automate this redirection so that if the visitor requests a ~/index.html URL it will change it to ~/index.php

As for the 500 Internal Server errors. I'll do as you say, enable debug mode and create a new issue on the OctoberCMS repository.

Thanks,
Matt.

**Comment by LukeTowers** [comment]
Are you using Apache and the .htaccess file? It's generally considered bad practice to have index.php in the URL in the first place. Also, don't create an issue until you have a specific error for people to take a look at, in fact lean towards asking questions on other platforms before making an issue for it.

As far as redirects go, I highly recommend purchasing the [Redirect plugin](https://octobercms.com/plugin/adrenth-redirect) and using that to setup redirects from the old site's urls.

**Comment by mjdwebdesign** [comment]
I believe it is using Apache and the .htaccess file. I haven't touched those so, anything to do with that is still set to the defaults that are setup with OctoberCMS.

Yeah, I can see what you mean about the bots. I'm fairly confident that is not the issue in this particular scenario though.

I did see the one you've linked earlier, I didn't look into with much detail due to the pricing. Initially, I've installed a free, basic redirect plugin ([here](https://octobercms.com/plugin/pmk-redirect)) to complete redirects for now.

I'll definitely look into the one you linked as a potential purchase as it might be more efficient for mass redirects.

We will see how things progress from here. If there appears to be no improvement after resolving these request logs with redirects I'll look to seek help from the forums rather than these repositories.

Thank you for your help Luke.
